code/modules/reagents/reagent_containers/dropper.dm

```diff
@@ -108,4 +108,16 @@
 	icon_state = "pipette"
 	amount_per_transfer_from_this = 1
 	possible_transfer_amounts = list(0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1)
-	volume = 1
\ No newline at end of file
+	volume = 1
+
+//Syndicate item. Virus transmitting mini hypospray
+/obj/item/reagent_containers/dropper/precision/viral_injector
+
+/obj/item/reagent_containers/dropper/precision/viral_injector/attack(mob/living/M, mob/living/user, def_zone)
+	if(M.can_inject(user, 1))
+		to_chat(user, "<span class='warning'>You stab [M] with the [src].</span>")
+		add_attack_logs(user, M, "Stabbed with Viral Injector")

```

Comment:
might just be easier/better to make it a hypospray subtype and add a stealth var?  🤷‍♂️

Reply:
The current hypospray does not infect people with viruses. This will mean that I need to change a lot of the code and add the dropper functionality.

Reply:
huh, weird, and also 

>If the admins want it I can add loging about which virus is injected. I tried to make this but failed at it. I can give it another try if people want me to.

just the reagent injected should be enough, since this can hold any reagent so people can inject with vodka or anything else

Reply:
I though about adding the viruses injected. Will this also help?